core/domain/skill_services.py

```diff
@@ -867,3 +867,31 @@ def skill_has_associated_questions(skill_id):
         question_models.QuestionSkillLinkModel.get_all_question_ids_linked_to_skill_id( # pylint: disable=line-too-long
             skill_id))
     return len(question_ids) > 0
+
+
+def filter_skills_by_mastery(user_id, skill_ids):
+    """Given a list of skill_ids, it returns a list of
+    20 (feconf.MAX_NUMBER_OF_SKILL_IDS) skill_ids in which the user has
+    the least mastery.(Please note that python 2.7 considers the None
+    type smaller than any value, so None types will be returned first)
```

Comment:
Wait, does that mean the returned list will have 'None' elements in it? If so, that shouldn't be the case.

Reply:
For the skills which don't have a UserSkillMasteryModel associated with them, this typically means that the user has not been associated with the skill, so the mastery for this skill type is returned as nil. We can take this as the user has no experience with that skill, and therefore has the least mastery in it. @sophiewu6 please correct me if I am wrong here.

Reply:
Yes that's correct.

Reply:
@Arnesh07 Yeah, that's correct, I meant whether the returned array from this function will have None elements? Though, from the looks of it, I don't think it does.

Reply:
@aks681 Okay, I misunderstood. This function only returns a list of skill_ids and they can not be None here, as you judged, so we are safe here.
Thanks!